src/portable/raspberrypi/rp2040/rp2040_usb.h

```diff
@@ -74,11 +90,24 @@ typedef struct hw_endpoint
 #endif
 } hw_endpoint_t;
 
+#if TUD_OPT_RP2040_USB_DEVICE_UFRAME_FIX
+extern volatile uint32_t last_sof;
+#endif
+
+uint32_t rp2040_critical_frame_period(struct hw_endpoint *ep);
```

Comment:
maybe consider making this a macro that returns 0 when !TUD_OPT_RP2040_USB_DEVICE_UFRAME_FIX rather than having an external function that always returns false.